cluster/cluster.go

```diff
@@ -29,65 +27,59 @@ func validMode(m string) bool {
 
 var (
 	Mode    ModeType
-	Manager *ClusterManager
-	cfg     *memberlist.Config
+	Manager ClusterManager
 	Tracer  opentracing.Tracer
 
 	InsufficientShardsAvailable = NewError(http.StatusServiceUnavailable, errors.New("Insufficient shards available."))
 )
 
 func Init(name, version string, started time.Time, apiScheme string, apiPort int) {
-	Manager = &ClusterManager{
-		members: map[string]Node{
-			name: {
-				Name:          name,
-				ApiPort:       apiPort,
-				ApiScheme:     apiScheme,
-				Started:       started,
-				Version:       version,
-				Priority:      10000,
-				Primary:       primary,
-				PrimaryChange: time.Now(),
-				StateChange:   time.Now(),
-				Updated:       time.Now(),
-				local:         true,
+	thisNode := Node{
+		Name:          name,
+		ApiPort:       apiPort,
+		ApiScheme:     apiScheme,
+		Started:       started,
+		Version:       version,
+		Primary:       primary,
+		Priority:      10000,
+		PrimaryChange: time.Now(),
+		StateChange:   time.Now(),
+		Updated:       time.Now(),
+		local:         true,
+	}
+	if Mode == ModeMulti {
+		mgr := &MemberlistManager{
+			members: map[string]Node{
+				name: thisNode,
 			},
-		},
-		nodeName: name,
+			nodeName: name,
+		}
+		mgr.cfg = memberlist.DefaultLANConfig()
+		mgr.cfg.BindPort = clusterPort
+		mgr.cfg.BindAddr = clusterHost.String()
+		mgr.cfg.AdvertisePort = clusterPort
+		mgr.cfg.Events = mgr
+		mgr.cfg.Delegate = mgr
+		h := sha256.New()
+		h.Write([]byte(ClusterName))
+		mgr.cfg.SecretKey = h.Sum(nil)
```

Comment:
doesn't this stuff better go into a MemberlistManager constructor?

Reply:
that would be better.